src/qibolab/instruments/zhinst.py

```diff
@@ -0,0 +1,615 @@
+                uid=sweeper.parameter.name,
+                values=sweeper.values,
+            )
+        return Zh_Sweeper
+
+    def add_sweeper(self, sweeper, qubit):
+        self.zhsweepers.append(self.select_sweeper(sweeper, sweeper.parameter.name, qubit))
+
+
+class Zurich(AbstractInstrument):
+    def __init__(self, name, descriptor, use_emulation=False):
+        self.name = name
+        self.descriptor = descriptor
+        self.emulation = use_emulation
+
+        self.is_connected = False
+
+        self.time_of_flight = 0
+        self.smearing = 0
+
+        self.signal_map = {}
+        self.calibration = lo.Calibration()
+
+        self.device_setup = None
+        self.session = None
+        self.device = None
+
+        self.relaxation_time = 0.0
+        self.time_of_flight = 0.0
+        self.smearing = 0.0
+
+        self.exp = None
+        self.experiment = None
+        self.results = None
+
+        self.sequence = None
+
+        # Improve the storing of multiple sweeps
+        self.nsweeps = 0.0
+        self.sweeps = None
+        self.sweepers = None
+
+    def connect(self):
+        if not self.is_connected:
+            for attempt in range(3):
+                try:
+                    self.device_setup = lo.DeviceSetup.from_descriptor(
+                        yaml_text=self.descriptor,
+                        server_host="localhost",
+                        server_port="8004",
+                        setup_name=self.name,
+                    )
+                    self.session = lo.Session(self.device_setup)
+                    self.device = self.session.connect(do_emulation=self.emulation)
+                    self.is_connected = True
+                    break
+                except Exception as exc:
+                    print(f"Unable to connect:\n{str(exc)}\nRetrying...")
+            if not self.is_connected:
+                raise InstrumentException(self, f"Unable to connect to {self.name}")
+
+    def start(self):
+        pass
+
+    def stop(self):
+        pass
+
+    def disconnect(self):
+        if self.is_connected:
+            self.device = self.session.disconnect()
+            self.is_connected = False
+        else:
+            print(f"Already disconnected")
+
+    def setup(self, qubits, relaxation_time=0, time_of_flight=0, smearing=0, **_kwargs):
+        self.relaxation_time = relaxation_time
+        self.time_of_flight = time_of_flight
+        self.smearing = smearing
+
+        self.signal_map = {}
+        self.calibration = lo.Calibration()
+
+        # Map can be done in experiment, but general calibration needs to be here
+        # General calibration != experiment calibration
+        for qubit in qubits.values():
+            if "c" in str(qubit.name):
+                self.register_flux_line(qubit)
+            else:
+                if qubit.flux == None:
+                    pass
+                else:
+                    self.register_flux_line(qubit)
```

Comment:
Removed the duplicated definition, I think they called it port_delay, but I would leave it as time_of_flight. I think we don't need smearing so I would remove it. I will apply the is not None to all the times I used something similar.

Reply:
After looking for it I think we should use smearing on all devices as it is not hard to implement. Following Yvonne Gao guide for superconducting qubits and following c) we should try to remove the start of the readout pulse from the acquisition and a smaller part before it ends to account for time of flight mismatch. 

![imagen](https://user-images.githubusercontent.com/19141117/227011473-f905e3ce-2fe2-4b90-a23b-c7c48e2796eb.png)

That would mean having a tuple for smearing instead of a float ? How do you use it in QM ?